src/inference/support/hmc_core.jl

```diff
@@ -66,42 +65,11 @@ function _hmc_step(θ::AbstractVector{<:Real},
     r = AHMC.rand_momentum(h)
     H = AHMC.hamiltonian_energy(h, θ, r)
 
-    # I have to copy https://github.com/TuringLang/AHMC.jl/blob/master/src/trajectory.jl#L24
-    # as the current interface doesn't return is_accept
-    H = AHMC.hamiltonian_energy(h, θ, r)
-    θ_new, r_new, is_valid = AHMC.step(prop.integrator, h, θ, r, prop.n_steps)
-    H_new = AHMC.hamiltonian_energy(h, θ_new, r_new)
-    # Accept via MH criteria
-    is_accept, α = AHMC.mh_accept(H, H_new)
-    if is_accept
-        θ, r = θ_new, -r_new
-    end
+    θ_new, r_new, α, H_new = AHMC.transition(prop, h, Vector{Float64}(θ), r)
+    # NOTE: as `transition` doesn't return `is_accept`, I use `H == H_new` as a check
+    is_accept = H != H_new  # If the new Hamiltonian enerygy is different
```

Comment:
Would it be safer to test `θ == θ_new` here, instead of `H != H_new`?

Reply:
I guess it will be safer if we doing so. I just worried about the performance of comparing two vectors. In fact, the Hamiltonian energy should be a good enough approximation (before we actually return the `is_accept` from AHMC.jl) as not only the model parameter should be the same to make the Hamiltonian energy be the same but also a random momentum variable should by luck be the same, which I guess will not happen in practise.